emgapi/metagenomics_exchange.py

```diff
@@ -0,0 +1,161 @@
+        analysis_registry_id = None
+        metadata_match = True
+
+        try:
+            response = self.get_request(endpoint=endpoint, params=params)
+        except:
+            logging.warning(f"Get API request failed")
+            return analysis_registry_id , metadata_match
+
+        if response.ok:
+            data = response.json()
+            datasets = data.get("datasets")
+            sourceIDs = [item.get("sourceID") for item in datasets]
+            if source_id in sourceIDs:
+                found_record = [item for item in datasets if item.get("sourceID") == source_id][0]
+                logging.info(f"{source_id} exists in ME")
+                analysis_registry_id  = found_record.get("registryID")
+                if metadata:
+                    for metadata_record in metadata:
+                        if not(metadata_record in found_record):
+                            metadata_match = False
+                            return analysis_registry_id , metadata_match
+                        else:
+                            if metadata[metadata_record] != found_record[metadata_record]:
+                                print(metadata[metadata_record], found_record[metadata_record])
+                                metadata_match = False
+                                logging.info(f"Incorrect field {metadata[metadata_record]} != {found_record[metadata_record]})")
+                                return analysis_registry_id, metadata_match
+                return analysis_registry_id , metadata_match
+            else:
+                logging.info(f"{source_id} does not exist in ME")
+        return analysis_registry_id, metadata_match
+
+    def delete_analysis(self, registry_id: str):
+        response = self.delete_request(endpoint=f"datasets/{registry_id}")
+        if response.ok:
+            logging.info(f"{registry_id} was deleted with {response.status_code}")
+            return True
+        else:
+            if response.status_code == 400:
+                logging.error(f"Bad request for {registry_id}")
+            elif response.status_code == 401:
+                logging.error(f"Failed to authenticate for {registry_id}")
+            else:
+                logging.error(f"{response.message} for {registry_id}")
+            return False
+
+    def patch_analysis(self, registry_id: str, data: dict):
+        response = self.patch_request(endpoint=f"datasets/{registry_id}", data=data)
+        if response.ok:
+            logging.info(f"{registry_id} was patched")
+            return True
+        else:
+            if response.status_code == 400:
+                logging.error(f"Bad request for {registry_id}")
+            elif response.status_code == 401:
+                logging.error(f"Fail to authenticate for {registry_id}")
+            elif response.status_code == 409:
+                logging.error(f"Conflicts with existing data for {registry_id}")
+            return False
```

Comment:
@mberacochea not really a big deal, 
but personally I find it easier to follow code when there aren't too many conditionals

Reply:
I like your suggestion, it improves the code